### ISSUE TITLE
Performance benchmark is need

### ISSUE THREAD

**Issue by lowang-bh** [volcano-sh/volcano#3244]
#### What would you like to be added:

1. some benchmark test for scheduling performance.

#### Why is this needed:

1. supply basic benchmark test, and avoid future's change has a negative effect
2. more easy to compare when do some performance improvement

**Comment by RamezesDong** [comment]
Hello, I have also been trying to add performance benchmarks for Volcano. The initial plan is that, to first construct a benchmark for a single action in the form of unit tests, building clusters of different scale magnitudes (100 nodes 500 pods; 1000 nodes 5000 pods; 1000 nodes 100000 pods 1000 jobs…). This will test the performance of a single action under various scenarios. In the following step, we will use the actual scheduler operation as a benchmark to evaluate the performance of Volcano from the start of an open session to the end of a close session.

Could you share some suggestions for me?

**Comment by lowang-bh** [comment]
Yes, your idea sound good and is corresponding to my opinion. There is already a mockSchedulerCache and UTframework commit in https://github.com/volcano-sh/volcano/pull/3269. We can add a performanceTest Framework base on the mockScheduler as the UTFramework. And then do as you said, we can mock various scale pods/nodes to test each actions and even the full-scheduling process with mockScheduler's `runOnce`.

**Comment by hwdef** [comment]
still need.

**Comment by tanjunchen** [comment]
still need.
